thejamesgore/Foxtons-Dashboard
# feat: Add new components and pages to enhance user experience

## Description 🚀

This pull request adds multiple new components and pages to the project, including:

- Common components: `Profile`, `PropertyCard`, and `CustomButton`.
- Chart components: `PieChart`, `PropertyReferrals`, and `TotalRevenue`.
- Agent components: `AgentCard`.
- Home components: `TopAgent`.
- Pages: `Login`, `Home`, `Agents`, `MyProfile`, `PropertyDetails`, `AllProperties`, `CreateProperty`, `AgentProfile`, and `EditProperty`.

These additions significantly enhance the user experience and provide additional functionality to the application, making it more robust and user-friendly.

## Changes Made 👉

- Created new files and directories for the components under `src/components`.
- Added new React components for the necessary interface, including forms, buttons, and tables.
- Updated existing React components to use the new interface components.

## Reason for Changes 🔍

By adding these new components and pages to our project, we can enhance the user experience and provide additional functionality for managing properties and tenants.

## How to Test

To test these changes, you can follow these steps:

1. Clone this branch to your local machine.
2. Install the necessary dependencies with `npm install`.
3. Start the application with `npm start`.
4. Verify that the application starts without any errors and that you can access the new pages by navigating to them using their respective paths.
5. Test the new components by interacting with the forms, buttons, and tables.

## Checklist


## Related Issues

None.

## Other Notes

None.
